oktapam/data_source_cloud_connections.go

```diff
@@ -0,0 +1,59 @@
+package oktapam
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+)
+
+func dataSourceCloudConnections() *schema.Resource {
+	return &schema.Resource{
+		Description: descriptions.SourceCloudConnections,
+		ReadContext: dataSourceCloudConnectionsList,
+		Schema: map[string]*schema.Schema{
+			attributes.Name: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.FilterName,
+			},
```

Comment:
nit: an option here would be to include the provider type, so a user can filter based on this.  not a huge issue to add this right now since we only have AWS connections, but would be something useful in the future.